tests/samplers/test_rejection_sampler.py

```diff
@@ -198,23 +213,86 @@ def test_deterministic_when_seeded(k: int, vocab_size: int, batch_size: int,
                 assert torch.equal(results[j][i], results[0][i])
 
 
+@pytest.mark.parametrize("k", [1, 3, 6])
+@pytest.mark.parametrize("vocab_size", [30_000, 50_000])
+@pytest.mark.parametrize("batch_size", [1, 8, 32, 128])
+@pytest.mark.parametrize("device", CUDA_DEVICES)
+@torch.inference_mode()
+def test_compare_nonflashinfer_backend(k: int, vocab_size: int,
+                                       batch_size: int, device: str):
+
+    def get_seeded_seqs():
+        seeded_mask = torch.rand(batch_size, dtype=torch.float32) <= 1.0
```

Comment:
I think this needs to go out of the helper function, else the rand will be different

Reply:
I realize there's an error here -- it should be `torch.rand(...) <= 0.5`

Reply:
I think I will just remove the rand, we should just fix the generator for each request in the batch instead of fixing it with 50% probability.